madanalysis/configuration/weight_configuration.py

```diff
@@ -0,0 +1,303 @@
+
+    @property
+    def aux(self) -> int:
+        """retreive aux value"""
+        return self._aux
+
+    @property
+    def dynamic_scale(self) -> int:
+        """retreive dynamic scale"""
+        return self._dyn_scale
+
+    @property
+    def muf(self) -> float:
+        """retreive factorisation scale"""
+        return self._muf
+
+    @property
+    def mur(self) -> float:
+        """retreive **forget the name** scale"""
+        return self._mur
+
+    @property
+    def pdfset(self) -> int:
+        """retreive pdf set id"""
+        return self._pdf
+
+    @property
+    def merging(self) -> float:
+        """retreive merging scale"""
+        return self._merging
+
+    @property
+    def is_nominal(self) -> bool:
+        """Return true if the weight is nominal"""
+        if all(
+            x is None
+            for x in [
+                self.aux,
+                self.pdfset,
+                self.dynamic_scale,
+                self.muf,
+                self.mur,
+                self.merging,
+            ]
+        ):
+            return True
+
+        return False
+
+
+class WeightCollection:
+    """Create a weight collection"""
+
+    def __init__(self, collection: List[Weight] = None):
+        self._collection = [] if collection is None else collection
+        self._names = []
+
+    def append(self, name: Text, idx: int) -> None:
+        """Add weight into the collection"""
+        if name not in self.names:
+            self._collection.append(Weight(name=name, loc=idx))
+
+    def __repr__(self) -> Text:
+        if len(self) < 5:
+            return "WeightCollection(" + ",".join([str(x) for x in self]) + ")"
+
+        return f"WeightCollection(contains {len(self)} weight definitions)"
+
+    def __str__(self) -> Text:
+        return self.__repr__()
+
+    def __iter__(self) -> Weight:
+        yield from self._collection
+
+    def __len__(self) -> int:
+        return len(self._collection)
+
+    def __getitem__(self, index: int) -> Weight:
+        return self._collection[index]
+
+    @property
+    def names(self) -> List[Text]:
+        """retreive weight names"""
+        if len(self) == len(self._names):
+            return self._names
+
+        self._names = [x.name for x in self]
+        return self._names
+
+    def to_dict(self) -> List[Dict[Text, Any]]:
+        """Convert to dictionary"""
+        return [x.to_dict() for x in self]
+
+    def from_dict(self, data: List[Dict[Text, Any]]) -> None:
+        """Construct collection from data"""
+        for dat in data:
+            self.append(dat["name"], dat["loc"])
+
+    @property
+    def nominal(self) -> Weight:
+        """Get nominal weight"""
+        for w in self:
+            if w.is_nominal:
+                return w
+        raise ValueError("Can not find nominal weight")
+
+    def group_for(self, group: Text) -> Dict:
+        """Create a group"""
+        assert group in ["muf", "mur", "pdfset", "dynamic_scale"]
```

Comment:
In principle, the dynamic scale weights could also be split in muf/mur. Why not doing it?

Reply:
This function can go away actually, I set many functions to create certain subgroups but ended up using just two of them